### PR TITLE
Remove memory store and increase test coverage

### PR DESCRIPTION
**Pull request by MHassanTariq**
## Description

This PR increases the test coverage. It adds test cases that were previously not catered. 

For details refer to issue #985 

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @rafaelcr or @zone117x for review

### CODE REVIEW

**Comment by MHassanTariq** [comment]
This PR is complete for now. I will make a new PR for rosetta test cases to increase the code coverage.

**Comment by MHassanTariq** [comment]
@rafaelcr I was trying to rebase this PR. However, one of my tests, [here](https://github.com/hirosystems/stacks-blockchain-api/pull/1011/files#diff-521b3ebceedf4b02e9f0a164d139b5b2cc65fbeae7c5a54d4f5ce265be2bb1e1R8865-R8868), that I added in this PR has been failing. The error shown is 
```
TypeError: Cannot read properties of undefined (reading 'count')
```
This error is generated as the length of rows for [total](https://github.com/hirosystems/stacks-blockchain-api/blob/develop/src/datastore/postgres-store.ts#L2974) is 0. 
Can you suggest anything on how to resolve this?

File: src/datastore/postgres-store.ts [hunk 1]
```diff
@@ -2985,6 +2986,7 @@ export class PgDataStore
         `,
         [limit, offset]
       );
+      console.log('printing results here: ', results);
```
Comment (rafaelcr) [inline src/datastore/postgres-store.ts:2989]: Could you remove these debug logs?

File: src/index.ts [hunk 2]
```diff
@@ -141,11 +140,6 @@ async function init(): Promise<void> {
     db = OfflineDummyStore;
   } else {
     switch (process.env['STACKS_BLOCKCHAIN_API_DB']) {
```
Comment (rafaelcr) [inline src/index.ts:142]: This env variable is now unused. Can you remove it from the entire project?

File: src/tests/setup.ts [hunk 3]
```diff
@@ -12,9 +12,11 @@ export default async (): Promise<void> => {
     process.env.NODE_ENV = 'test';
   }
   loadDotEnv();
+  process.env.PG_DATABASE = 'postgres';
+  const db = await PgDataStore.connect({ skipMigrations: true, usageName: 'setup' });
   const server = await startEventServer({
```
Comment (rafaelcr) [inline src/tests/setup.ts:17]: If I'm not mistaken, this entire global server is never used... Can you try removing it (also removing the db) from here and `teardown.ts` to see if tests are at all affected? If not, we should delete it.

**Review by rafaelcr — CHANGES_REQUESTED** [review]

File: src/tests-bns/setup.ts [hunk 4]
```diff
@@ -1,24 +1,11 @@
...
-export interface GlobalServices {
-  db: PgDataStore;
-}
 
 // ts-unused-exports:disable-next-line
 export default async (): Promise<void> => {
   console.log('Jest - setup..');
-  if (!process.env.NODE_ENV) {
-    process.env.NODE_ENV = 'test';
-  }
```
Comment (rafaelcr) [inline src/tests-bns/setup.ts:14]: These lines should be kept

File: src/tests-rosetta/setup.ts [hunk 5]
```diff
@@ -1,24 +1,11 @@
...
-export interface GlobalServices {
-  db: PgDataStore;
-}
 
 // ts-unused-exports:disable-next-line
 export default async (): Promise<void> => {
   console.log('Jest - setup..');
-  if (!process.env.NODE_ENV) {
-    process.env.NODE_ENV = 'test';
-  }
```
Comment (rafaelcr) [inline src/tests-rosetta/setup.ts:14]: Keep these lines too

File: src/tests-tokens/setup.ts [hunk 6]
```diff
@@ -1,24 +1,7 @@
...
-  db: PgDataStore;
-}
-
 // ts-unused-exports:disable-next-line
-export default async (): Promise<void> => {
+export default (): void => {
   console.log('Jest - setup..');
-  if (!process.env.NODE_ENV) {
-    process.env.NODE_ENV = 'test';
-  }
```
Comment (rafaelcr) [inline src/tests-tokens/setup.ts:14]: Keep this

File: utils/src/index.ts [hunk 7]
```diff
@@ -27,7 +27,7 @@ type TableCellValue = string | number | bigint | undefined;
  * @param blockHeight Specific block height at which to query balances
  */
 async function printTopAccountBalances(count: number, blockHeight: number) {
-  const db = await PgDataStore.connect(true);
+  const db = await PgDataStore.connect({ skipMigrations: true, usageName: 'tests' });
```
Comment (rafaelcr) [inline utils/src/index.ts:30]:
```suggestion
  const db = await PgDataStore.connect({ skipMigrations: true, usageName: 'utils' });
```

**Review by rafaelcr — CHANGES_REQUESTED** [review]

**Review by rafaelcr — APPROVED** [review]
LGTM 👍 
Before merging, can you change the PR's title to "Remove memory store and increase test coverage"?

**Review by rafaelcr — APPROVED** [review]

**Pull request merged**